src/routers/product.router.ts

```diff
@@ -2,21 +2,45 @@ import express, { Request, Response } from "express";
 import { ProductV2Service } from "../services/productV2.service";
 import { IProduct } from "../services/product.interface";
 import { errorHandling } from "./error-handling";
+import { IPagination } from "./pagination.interface";
 
 export const productRouter: express.Router = express.Router();
 const productV2Service = new ProductV2Service();
 
 productRouter.get("", async (req: Request, res: Response) => {
   try {
     const productList = await productV2Service.getAllProducts();
-    res.status(200).json(productList);
+    // Pagination
+    const pagination: IPagination = {};
+    const page: any = req.query.page;
```

Comment:
why any?

Reply:
because If i put "number" i get the following error:
Type 'string | ParsedQs | string[] | ParsedQs[] | undefined' is not assignable to type 'number'.
  Type 'undefined' is not assignable to type 'number'.

Reply:
not sure how to bypass it

Reply:
I think you should start to do some kind of input validation to all the API s.  usung middleware functions.  I think there's is a very nice project over the web for middleware routers validation.
Do the research and come back with what did you found, and what you considered should be the best one.